src/routes/(console)/organization-[organization]/billing/planSummary.svelte

```diff
@@ -1,180 +1,564 @@
+        ) {
+            return [];
+        }
+
+        const max = typeof maxValue === 'string' ? parseFloat(maxValue) : maxValue;
+        if (max <= 0) return [];
+
+        const percentage = Math.min((currentValue / max) * 100, 100);
+        const progressColor = getProgressColor(percentage);
+
+        return [
+            {
+                size: currentValue,
+                color: progressColor,
+                tooltip: {
+                    title: `${percentage.toFixed(1)}% used`,
+                    label: `${currentValue.toLocaleString()} of ${max.toLocaleString()}`
+                }
+            }
+        ];
+    }
+
+    function createStorageProgressData(
+        currentBytes: number,
+        maxGB: number
+    ): Array<{ size: number; color: string; tooltip?: { title: string; label: string } }> {
+        if (maxGB <= 0) return [];
+
+        const maxBytes = maxGB * 1024 * 1024 * 1024;
+        const percentage = Math.min((currentBytes / maxBytes) * 100, 100);
+        const progressColor = getProgressColor(percentage);
+
+        const currentSize = humanFileSize(currentBytes);
+
+        return [
+            {
+                size: currentBytes,
+                color: progressColor,
+                tooltip: {
+                    title: `${percentage.toFixed(0)}% used`,
+                    label: `${currentSize.value} ${currentSize.unit} of ${maxGB} GB`
+                }
+            }
+        ];
+    }
+
+    $: projectsList = (() => {
+        // Get all projects from usageProjects (which now contains all organization projects)
+        const allProjectIds = Object.keys(usageProjects);
+
+        if (allProjectIds.length === 0) {
+            return [];
+        }
+
+        return allProjectIds.map((projectId) => {
+            // Find the corresponding project data from organizationUsage
+            const projectData = organizationUsage?.projects?.find(
+                (proj: any) => proj.projectId === projectId
+            );
+
+            // Find the corresponding breakdown data
+            const breakdownData = currentAggregation?.projectBreakdown?.find(
+                (p: any) => p.$id === projectId
+            );
+
+            return {
+                projectId: projectId,
+                storage: projectData?.storage || 0,
+                executions: projectData?.executions || 0,
+                executionsMBSeconds: projectData?.executionsMBSeconds || 0,
+                bandwidth: projectData?.bandwidth || 0,
+                databasesReads: projectData?.databasesReads || 0,
+                databasesWrites: projectData?.databasesWrites || 0,
+                users: projectData?.users || 0,
+                authPhoneTotal: projectData?.authPhoneTotal || 0,
+                authPhoneEstimate: projectData?.authPhoneEstimate || 0,
+                breakdown: breakdownData
+            };
+        });
+    })();
+
+    $: billingData = [
```

Comment:
extract to a function and the do `$billingData = getBillingData(...)` Also the logic can be streamlined.